beancount/parser/booking_full_test.py

```diff
@@ -1726,7 +1783,42 @@ def test_reduce__missing_units_number(self, _, __):
         2016-01-01 * #booked
         """
 
+    @book_test(Booking.STRICT)
+    def test_reduce__star_merging(self, _, __):
+        """
+        2016-01-01 * #ante
+          Assets:Account          10 HOOL {10.00 USD, 2016-01-01}
+          Assets:Account          10 HOOL {20.00 USD, 2016-04-15}
+
+        2016-05-02 * #apply
+          Assets:Account          -5 HOOL {*}
+
+        2016-05-02 * #apply
+          Assets:Account          -5 HOOL {15 USD, *}
+
+        2016-05-02 * #booked
+          Assets:Account          -10 HOOL {10.00 USD, 2016-01-01}
+          Assets:Account          -10 HOOL {20.00 USD, 2016-04-15}
+          Assets:Account          20 HOOL {15.00 USD, 2016-01-01}
+          Assets:Account          -5 HOOL {15.00 USD, 2016-01-01}
 
+        2016-05-02 * #ex
+          Assets:Account          15 HOOL {15.00 USD, 2016-01-01}
+        """
+
+    @book_test(Booking.STRICT)
+    def test_reduce__star_merging_nomatch(self, _, __):
+        """
+        2016-01-01 * #ante
+          Assets:Account          10 HOOL {10.00 USD, 2016-01-01}
+          Assets:Account          10 HOOL {20.00 USD, 2016-04-15}
+
+        2016-05-02 * #apply
+          Assets:Account          -5 HOOL {99 USD, *}
+
+        2016-05-02 * #booked
+          error: "No position matches"
```

Comment:
Wellll that's an interesting case. We could allow an arbitrary amount to be deducted from the average cost when reducing. Not absolutely necessary,  but that could make sense.

Do you have a test for a pure cost basis adjustment? How would that look like?

Reply:
I thought carefully about allowing an arbitrary amount to be deducted from the cost basis when reducing (thus repricing the remaining ante-inventory), especially since I saw several test cases indicating that's the behavior you had in mind when you put the test cases together. Unless I'm missing a compelling usecase for it, it seems much better to check the basis is close to our average basis instead.
I rewrote the original section from my google doc discussing this to be much more clear. Would you take a look at just that section and let me know if you still want it to work this way? https://docs.google.com/document/d/1rclTmxJzq-4Na1UPY0XfS9Yzuv_SuHpdfHtabNsUH0E/edit#bookmark=id.qd80c26c1n8u
Sections: "Using basis to as correctness verification", and "Cost basis adjustment - leads to confusing ante-inventory"
Thanks!

Reply:
I'm not sure what you mean by "pure cost basis adjustment", When would someone do that? Is it just someone adding a transaction that manually removes all the holdings at a particular price and re-adds them with a different basis?